tests/mr/device/arena_mr_tests.cpp

```diff
@@ -533,6 +533,42 @@ TEST_F(ArenaTest, Defragment)  // NOLINT
   }());
 }
 
+TEST_F(ArenaTest, PerThreadAndStreamArenas)  // NOLINT
+{
+  auto const arena_size = superblock::minimum_size * 4;
+  arena_mr mr(rmm::mr::get_current_device_resource(), arena_size);
+  std::vector<std::thread> threads;
+  std::size_t num_threads{3};
+  auto view        = std::make_shared<rmm::cuda_stream_view>(rmm::cuda_stream_per_thread);
+  void* thread_ptr = mr.allocate(256, view->value());
+  threads.reserve(num_threads);
+  for (std::size_t i = 0; i < num_threads; ++i) {
+    threads.emplace_back(std::thread([&] {
+      cuda_stream stream{};
+      void* ptr = mr.allocate(32_KiB, stream);
```

Comment:
These all allocate in a stream arena (not PTDS). 

And, because the superblock size is 1MiB, all three allocations end up in the same (single) superblock, which is the same superblock as that which allocated `thread_ptr`. (I think).

Reply:
I am simplifying the test, I don't actually need 3 stream arena allocations. Also updated the comments so hopefully that answers your questions